src/sentry/integrations/vsts/client.py

```diff
@@ -72,49 +78,108 @@ class VstsApiPath:
+        self.oauth_redirect_url = oauth_redirect_url
+        self.access_token = access_token
+
+    def request(self, method, path, data=None, params=None, api_preview: bool = False):
+        headers = prepare_headers(
+            access_token=self.access_token,
+            api_version=self.api_version,
+            method=method,
+            api_version_preview=self.api_version_preview if api_preview else "",
+        )
+        return self._request(method, path, headers=headers, data=data, params=params)
+
+
+class VstsApiClient(IntegrationProxyClient, OAuth2RefreshMixin, VstsApiMixin):
     integration_name = "vsts"
+    _identity: Identity | None = None
 
     def __init__(
-        self, identity: "Identity", oauth_redirect_url: str, *args: Any, **kwargs: Any
+        self,
+        base_url: str,
+        oauth_redirect_url: str,
+        org_integration_id: int,
+        identity_id: int | None = None,
     ) -> None:
-        super().__init__(*args, **kwargs)
-        self.identity = identity
+        self.base_url = base_url
+        self.identity_id = identity_id
         self.oauth_redirect_url = oauth_redirect_url
-        if "access_token" not in self.identity.data:
-            raise ValueError("Vsts Identity missing access token")
+        super().__init__(org_integration_id=org_integration_id)
+
+    @property
+    def identity(self):
+        if self._identity:
+            return self._identity
+        self._identity = Identity.objects.filter(id=self.identity_id).first()
+        return self._identity
 
-    def request(
+    @control_silo_function
+    def authorize_request(
         self,
-        method: str,
-        path: str,
-        data: Optional[Mapping[str, Any]] = None,
-        params: Optional[Sequence[Any]] = None,
+        prepared_request: PreparedRequest,
         api_preview: bool = False,
-        timeout: Optional[int] = None,
-    ) -> Response:
+    ) -> PreparedRequest:
         self.check_auth(redirect_url=self.oauth_redirect_url)
-        headers = {
-            "Accept": "application/json; api-version={}{}".format(
-                self.api_version, self.api_version_preview if api_preview else ""
-            ),
-            "Content-Type": "application/json-patch+json"
-            if method == "PATCH"
-            else "application/json",
-            "X-HTTP-Method-Override": method,
-            "X-TFS-FedAuthRedirect": "Suppress",
-            "Authorization": "Bearer {}".format(self.identity.data["access_token"]),
-        }
-        return self._request(
-            method, path, headers=headers, data=data, params=params, timeout=timeout
+        method = prepared_request.method
+        access_token = self.identity.data["access_token"]
+        headers = prepare_headers(
+            access_token=access_token,
+            api_version=self.api_version,
+            method=method,
+            api_version_preview=self.api_version_preview if api_preview else "",
         )
```

Comment:
I think we can get around haing the `**kwargs` as params for this method if we just keep `def request()` and have that function set the `Accept` header. It's totally fine that that header be set on the region silo since it doesn't have auth data at all and It'd let us keep the signature for `authorize_request` and `finalize_request` strict.

Reply:
```python
def request(self, method, path, data, params, api_preview, timeout):
    accept_header = "application/json; api-version={}{}".format(self.api_version, self.api_version_preview if api_preview else ""

    return self._request(
       method, path, headers={"Accept": accept_header}, data=data, params=params, timeout=timeout
    )
```
You'd still need authorize_request for the rest of the sensitive headers though